addons/mrp/models/stock_move.py

```diff
@@ -242,6 +243,40 @@ def default_get(self, fields_list):
                 defaults['reference'] = production_id.name
         return defaults
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        """ Enforce consistent values (i.e. match _get_move_raw_values/_get_move_finished_values) for:
+        - Manually added components/byproducts specifically values we can't set via view with "default_"
+        - Moves from a copied MO
+        - Backorders
+        """
+        mo_id_to_mo = defaultdict(lambda: self.env['mrp.production'])
+        product_id_to_product = defaultdict(lambda: self.env['product.product'])
+        for value in vals_list:
+            mo_id = value.get('raw_material_production_id', False) or value.get('production_id', False)
+            if mo_id:
+                mo = mo_id_to_mo[mo_id]
+                if not mo:
+                    mo = mo.browse(mo_id)
+                    mo_id_to_mo[mo_id] = mo
+                value['name'] = mo.name
+                value['origin'] = mo._get_origin()
+                value['group_id'] = mo.procurement_group_id.id
+                value['propagate_cancel'] = mo.propagate_cancel
+                if value.get('raw_material_production_id', False):
+                    product = product_id_to_product[value['product_id']]
+                    if not product:
+                        product = product.browse(value['product_id'])
+                    product_id_to_product[value['product_id']] = product
+                    value['location_dest_id'] = mo.production_location_id.id
+                    value['price_unit'] = product.standard_price
+                    continue
+                # produced products + byproducts
+                value['location_id'] = mo.production_location_id.id
+                value['date'] = mo._get_date_planned_finished()
+                value['date_deadline'] = mo.date_deadline
+        return super().create(vals_list)
```

Comment:
Also I'm doing this PR https://github.com/odoo-dev/odoo/commit/c56c5fc569523bc9882baea392d5dfac6026c8c2 I'm not sure that this part will be needed in the future

Reply:
Some of the use cases will be fixed by the switch from onchange to compute, but I don't think all of them will, specifically when a component/byproduct is added to a non-draft MO and maybe also during MO splits (since the moves are created directly by the method rather than the compute). This is based on the PR as it current is of course.

Side note, I noticed your PR also changes the default `name` of the (finished+raw) moves, so this will probably need to be updated as part of your PR.